hydra-access-controls/lib/hydra/policy_aware_access_controls_enforcement.rb

```diff
@@ -1,36 +1,36 @@
 # Repeats access controls evaluation methods, but checks against a governing "Policy" object (or "Collection" object) that provides inherited access controls.
 module Hydra::PolicyAwareAccessControlsEnforcement
 
-  # Extends Hydra::AccessControlsEnforcement.apply_gated_discovery to reflect policy-provided access
-  # appends the result of policy_clauses into the :fq
-  # @param solr_parameters the current solr parameters
-  # @param user_parameters the current user-subitted parameters
+  # Extends Hydra::AccessControlsEnforcement.apply_gated_discovery to reflect policy-provided access.
+  # Appends the result of policy_clauses into the :fq
+  # @param [Hash] solr_parameters the current solr parameters, to be modified herein!
   def apply_gated_discovery(solr_parameters)
     super
     logger.debug("POLICY-aware Solr parameters: #{ solr_parameters.inspect }")
   end
 
-  # returns solr query for finding all objects whose policies grant discover access to current_user
+  # @return [String,void] solr query for finding all objects whose policies grant discover access to current_user
```

Comment:
Right -- my understanding is `void` is meant for incidental return values to advertise so that consumers (who happen to read the documentation) don't depend on any particular thing coming back. Here, `nil` seems to be part of the contract.